qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -98,11 +122,29 @@ def run(self, dag):
         )
 
         # Search for a path from source to target basis.
-
         search_start_time = time.time()
         basis_transforms = _basis_search(
             self._equiv_lib, source_basis, target_basis, _basis_heuristic
         )
+
+        qarg_local_basis_transforms = {}
+        for qarg, extra_source_basis in incomplete_source_basis.items():
+            expanded_target = target_basis | qarg_with_incomplete[qarg]
+            if len(qarg) > 1:
+                for qubit in qarg:
+                    expanded_target |= qarg_with_incomplete.get((qubit,), set())
```

Comment:
The iteration here should be done over all subsets of `qarg`. For example, if `len(qarg) = 3`, we should take gates acting on single qubits *and* gates acting on all pairs of qubits in `qargs`.

Reply:
Ah, yeah good catch. I fixed this in: https://github.com/Qiskit/qiskit-terra/pull/7227/commits/7e8adf47754279f05a9aefe4ad8b0a1232586143